substrate/frame/message-queue/src/lib.rs

```diff
@@ -1554,13 +1566,20 @@ impl<T: Get<O>, O: Into<u32>> Get<u32> for IntoU32<T, O> {
 impl<T: Config> ServiceQueues for Pallet<T> {
 	type OverweightMessageAddress = (MessageOriginOf<T>, PageIndex, T::Size);
 
-	fn service_queues(weight_limit: Weight) -> Weight {
+	fn service_queues(weight_limit: Weight, context: ServiceQueuesContext) -> Weight {
 		let mut weight = WeightMeter::with_limit(weight_limit);
 
 		// Get the maximum weight that processing a single message may take:
 		let max_weight = Self::max_message_weight(weight_limit).unwrap_or_else(|| {
-			defensive!("Not enough weight to service a single message.");
-			Weight::zero()
+			// throw defensive message when service_queues is called from on_initialize
+			// don't throw message when service_queues is called from on_idle
+			match context {
+				ServiceQueuesContext::OnInitialize => {
+					defensive!("Not enough weight to service a single message.");
+					Weight::zero()
+				},
+				ServiceQueuesContext::OnIdle => Weight::zero(),
+			}
```

Comment:
```suggestion
			if matches!(context, ServiceQueuesContext::OnInitialize) {
				defensive!("Not enough weight to service a single message.");
			}

			Weight::zero()
```

Reply:
Cleaner, less verbose. Thanks.

Reply:
@bkchr 

Does it make sense to replace the `ServiceQueuesContext` enum with a unit-like struct since we're now only using `OnInitialize`? There's no need for `OnIdle` variant after your refactoring.

Reply:
The enum is just for better readability. Before you could already have used a bool, but the enum makes it more clear and easier to understand.

Reply:
Okay cool

Reply:
@gotnoshoeson did you forgot to push your code?

Reply:
Yeah, will push now.